### DISCUSSION TITLE
Are decorators run during build time (npm run build-storybook)?

### DISCUSSION THREAD

**Discussion by rafaelrozon** [storybookjs/storybook#20696] [Help]
Hello,

I wrote a decorator that adds a parameter to selected stories based on their names. I do this in the preview.js file because I want to apply this parameter globally. When I launch Storybook and I log the stories I see that the injected parameter is there. However, when I build the storybook to a folder and then inspect the output files, there's no sign of the injected parameter. Why is that? Are decorators expected to run during build (`npm run build-storybook`)?

PS: What I'm really trying to do is to make Chromatic ignore certain stories, but I don't want to add `chromatic: { disable: true }`, to the stories because 1) I want a future-proof solution and 2) I have a bunch of stories to update (+200).

Thanks!

**Comment by sheriffMoose** [comment]
Technically, decorators are ran during runtime not build time. This is the intended behavior as they are ran through the preview iframe, meaning they run for each story when they are loaded through the UI.
